### PR TITLE
Add process_request_headers/3 to pass options

### LINKED ISSUE #129: process_request_headers doesn't have access to options
We would like to inject additional headers depending on options. In particular we have the following [extract from here](https://github.com/apache/couchdb/blob/master/test/elixir/lib/couch.ex#L62:L79)
```
def process_request_headers(headers, options) do
    case Keyword.get(options, :cookie) do
      nil ->
        headers
      cookie ->
        Keyword.put(headers, :Cookie, cookie)
    end
end
```

Unfortunately in order to achieve that we have to [override](https://github.com/apache/couchdb/blob/master/test/elixir/lib/couch.ex#L167:L228) `process_arguments/3` function. Which is quite complex. 

I've noticed that recently the https://github.com/myfreeweb/httpotion/pull/128 was merged. Which modifies the signature of process_request_headers to support passing body. 
Would it be possible to use similar approach and implement
```
def process_request_headers(headers, _body, _options), do: process_request_headers(headers)
```

```
@@ -146,7 +148,7 @@ defmodule HTTPotion.Base do
        headers    = Application.get_env(:httpotion, :default_headers, [])
                     |> Keyword.merge(Keyword.get(options, :headers, [])
                       |> Enum.map(fn ({k, v}) -> { (if is_atom(k), do: k, else: String.to_atom(to_string(k))), to_string(v) } end))
-                     |> process_request_headers(body)
+                     |> process_request_headers(body, options)
```

### PR DESCRIPTION
**Pull request by iilyak**
This change passes provided options from process_arguments/3 to
process_request_headers. This enables the user to inject custom
headers based on options. The change is backward compatible.

Fixes #129

### CODE REVIEW

File: lib/httpotion.ex [hunk 1]
```diff
@@ -57,6 +57,8 @@ defmodule HTTPotion.Base do
 
       def process_request_headers(headers, _body), do: process_request_headers(headers)
 
+      def process_request_headers(headers, _body, _options), do: process_request_headers(headers)
```
Comment (myfreeweb) [inline lib/httpotion.ex]: Call with `(headers, body)` here — overriding the `/2` version should still work

Comment (iilyak) [reply lib/httpotion.ex]: Fixed in https://github.com/myfreeweb/httpotion/pull/130/commits/d624f496f0218e1f94f4787d3421b907ac3edd9c. I'll squash commits after approval.

Comment (myfreeweb) [reply lib/httpotion.ex]: no need to, I'll squash using github

**Comment by myfreeweb** [comment]
Thanks!

**Comment by iilyak** [comment]
@myfreeweb Would you be able to push a release with this fix?

**Comment by myfreeweb** [comment]
uploaded 3.1.3

**Comment by iilyak** [comment]
thank you very much.

**Pull request merged**